src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -1486,6 +1486,17 @@ ntp_archiver::maybe_upload_aborted_tx(
     co_return std::nullopt;
 }
 
+ss::future<> ntp_archiver::upload_index(
+  ss::sstring path, cloud_storage::offset_index index) {
+    retry_chain_node rtc{
+      _conf->segment_upload_timeout(),
+      100ms,
+      retry_strategy::disallow,
+      &_rtcnode};
+    std::ignore = co_await _remote.upload_index(
```

Comment:
this could throw in case of timeout, I think it makes sense to handle the exception or use `coroutine::as_future`

Reply:
oh yeah of course. good catch, thanks.

Reply:
done